### PR TITLE
EHPR-108 | Fix DB connection issue

### PR DESCRIPTION
**Pull request by fwpushan**
- [x] Uncouple nest application config code from bootstrap code

### CODE REVIEW

**Review by arranfw — APPROVED** [review]
Interesting, just changing the file fixes the issue? Could you explain what the problem was?

**Comment by hitankar-freshworks** [comment]
I'd be surprised if changing the file would work. I was thinking in the line of our TF config. Maybe VPC?

**Review by arranfw — CHANGES_REQUESTED** [review]
If the file main->config file change isn't needed lets revert it

**Review by sunghwan-park-fw — APPROVED** [review]

File: apps/api/src/.ormconfig.ts [hunk 1]
```diff
@@ -4,7 +4,6 @@ import { DatabaseNamingStrategy } from './database/database.naming-strategy';
 dotenv.config();
 // Check typeORM documentation for more information.
 export const config: ConnectionOptions = {
-  name: 'default',
```
Comment (hitankar-freshworks) [inline apps/api/src/.ormconfig.ts:7]: I don't think this is the issue. perhaps we can set `connectTimeoutMS` to `10000`

File: apps/api/src/.ormconfig.ts [hunk 2]
```diff
@@ -4,7 +4,6 @@ import { DatabaseNamingStrategy } from './database/database.naming-strategy';
 dotenv.config();
 // Check typeORM documentation for more information.
 export const config: ConnectionOptions = {
```
Comment (hitankar-freshworks) [inline apps/api/src/.ormconfig.ts:6]:
```suggestion
export const config: PostgresConnectionOptions = {
```

File: apps/api/src/main.ts [hunk 3]
```diff
@@ -1,92 +1,4 @@
...
-
-  return {
-    app,
-    expressApp,
-  };
-}
+import { createNestApp } from './app.config';
 
 async function bootstrap() {
   const { app, ...other } = await createNestApp();
```
Comment (hitankar-freshworks) [inline apps/api/src/main.ts:4]: other is not needed

**Review by hitankar-freshworks — CHANGES_REQUESTED** [review]
added couple of suggestions

**Comment by fwpushan** [comment]
> If the file main->config file change isn't needed lets revert it

@arranfw  There are multiple failure on the code de-coupling solve one of them.

**Comment by vyasworks** [comment]
> > If the file main->config file change isn't needed lets revert it
> 
> @arranfw There are multiple failure on the code de-coupling solve one of them.

can you please list them and what each one solves?

File: apps/api/src/main.ts [hunk 4]
```diff
@@ -1,92 +1,4 @@
...
-  });
-
-  return {
-    app,
-    expressApp,
-  };
-}
+import { createNestApp } from './app.config';
 
 async function bootstrap() {
```
Comment (fwpushan) [inline apps/api/src/main.ts:3]: @arranfw @hitankar-freshworks @dbayly-freshworks @vyasworks 
Decoupling bootstrap code from nest app config code **Fix multiple default connection issue**

File: apps/api/src/form/entity/form.entity.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { FormPayloadDTO } from '@ehpr/common';
 @Entity('form')
 export class FormEntity extends BaseEntity {
   @Exclude()
-  @Column('jsonb', { nullable: false })
+  @Column('jsonb', { nullable: true })
```
Comment (fwpushan) [inline apps/api/src/form/entity/form.entity.ts:9]: @arranfw @hitankar-freshworks @vyasworks @dbayly-freshworks 
This is not required we can revert this.

File: apps/api/src/form/form.service.ts [hunk 6]
```diff
@@ -24,15 +26,22 @@ export class FormService {
...
+    if (!newForm.version) {
+      newForm.version = '1.0.0';
+    }
     const savedForm = await this.formRepository.save(newForm);
-
-    const notifiedForm = await this.sendMail(savedForm);
-
-    return notifiedForm;
+    this.logger.log(`Saved form with id ${savedForm.id}`);
+    if (savedForm.payload?.contactInformation.email) {
```
Comment (fwpushan) [inline apps/api/src/form/form.service.ts:34]: @arranfw @hitankar-freshworks @dbayly-freshworks @vyasworks 
Any improper payload can break API, so this defensive code is introduce .... but no effect if validation enables in system

File: apps/api/src/mail/mail.service.ts [hunk 7]
```diff
@@ -30,48 +29,43 @@ export class MailService {
...
-          auth: {
-            username: process.env.CHES_CLIENT_ID as string,
-            password: process.env.CHES_CLIENT_SECRET as string,
-          },
-          headers: {
-            'Content-Type': 'application/x-www-form-urlencoded',
-          },
+    const token = await axios.post(
+      process.env.CHES_AUTH_URL as string,
+      stringify({ grant_type: 'client_credentials' }),
```
Comment (fwpushan) [inline apps/api/src/mail/mail.service.ts:56]: @vyasworks @hitankar-freshworks @arranfw @dbayly-freshworks 
some unknown issue is occurred regarding usage of `URLSearchParams` to construct request body. May be that is not well supported by Node.

Comment (arranfw) [reply apps/api/src/mail/mail.service.ts:56]: This is surprising, URLSearchParams looks fully supported and has worked well in the past, maybe we can revisit this implementation before opting for a library over native functionality

File: apps/api/src/form/form.service.ts [hunk 8]
```diff
@@ -24,15 +26,22 @@ export class FormService {
       confirmationId,
     } as Partial<FormEntity>);
 
+    if (!newForm.version) {
+      newForm.version = '1.0.0';
```
Comment (arranfw) [inline apps/api/src/form/form.service.ts:30]: Should we default assume un-specified submissions are 1.0.0? This also doesn't fit the format I've seen previously, I thought the value was `v1`

File: apps/api/src/form/form.service.ts (hunk 6 above)
Comment (arranfw) [reply apps/api/src/form/form.service.ts:34]: I'd be comfortable with a TODO here, we don't want this staying past go-live

**Comment by fwpushan** [comment]
@hitankar-freshworks @vyasworks @arranfw @dbayly-freshworks Please check my comments on code.

File: apps/api/src/form/form.service.ts (hunk 8 above)
Comment (fwpushan) [reply apps/api/src/form/form.service.ts:30]: Again this is defensive code when validation is not present

File: apps/api/src/mail/mail.service.ts (hunk 7 above)
Comment (fwpushan) [reply apps/api/src/mail/mail.service.ts:56]: `qs` is node js native lib

Comment (arranfw) [reply apps/api/src/mail/mail.service.ts:56]: Egg on my face!

**Review by arranfw — APPROVED** [review]

**Review by arranfw — APPROVED** [review]

**Pull request merged**